### PR TITLE
Creates new projects inside a same-named folder

### LINKED ISSUE #10974: SQL Project should create folder for new project
In the SQL project, creating new project creates a new file. I was expecting it to be a folder so that new items will be added to same folder. I didn't know that so I was accidentally adding too many project files to the base folder I selected when creating the project

### PR DESCRIPTION
**Pull request by Benjin**
Fixes #10974 by creating a new folder (with the same name as the project) inside the directory the user picked during the new project workflow.

### CODE REVIEW

File: extensions/sql-database-projects/src/test/projectController.test.ts [hunk 1]
```diff
@@ -166,7 +166,7 @@ describe.skip('ProjectsController: project controller operations', function ():
 			let profilePath = await testUtils.createTestFile(baselines.publishProfileBaseline, 'publishProfile.publish.xml');
 			const projController = new ProjectsController(testContext.apiWrapper.object, new SqlDatabaseProjectTreeViewProvider());
 
-			let result = await projController.readPublishProfile(vscode.Uri.parse(profilePath));
+			let result = await projController.readPublishProfile(vscode.Uri.file(profilePath));
```
Comment (kisantia) [inline extensions/sql-database-projects/src/test/projectController.test.ts:169]: is it better to use Uri.file() over Uri.parse()?

Comment (Benjin) [reply extensions/sql-database-projects/src/test/projectController.test.ts:169]: For file paths, yeah - it accounts for Windows- and Unix-style paths together, while `.parse()` expects things to all use forward slashes (better for treeItem URIs or web URLs... or anything other than Windows filesystem paths, lol).

As is, the test in a slight corner-case: my repo is on `F:\`, but the test files get created in the user temp folder, which is on `C:\`.  When the URI gets read with `.parse()`, it ends up looking in `F:\path\to\temp\profile.xml` instead of `C:\path\to\temp\profile.xml`
